src/backend/x11/mod.rs

```diff
@@ -367,7 +367,10 @@ impl X11Handle {
             return Err(X11Error::InvalidWindow);
         }
 
-        let modifiers = modifiers.collect::<Vec<_>>();
+        let mut modifiers = modifiers.collect::<Vec<_>>();
+        if !(window.0.extensions.dri3 >= Some((1, 2))) {
```

Comment:
Make sure you put a comment here indicating why we limit the modifiers here, the content of the commit message should be fine.